test/core/security/grpc_crl_provider_test.cc

```diff
@@ -28,23 +28,37 @@
 #include <grpc/grpc_audit_logging.h>
 #include <grpc/grpc_crl_provider.h>
 
+#include "src/core/lib/security/credentials/tls/grpc_tls_crl_provider.h"
 #include "test/core/util/test_config.h"
 #include "test/core/util/tls_utils.h"
 
 #define CRL_PATH "test/core/tsi/test_creds/crl_data/crls/ab06acdd.r0"
+#define CRL_ISSUER "/C=AU/ST=Some-State/O=Internet Widgits Pty Ltd/CN=testca"
 
 namespace grpc_core {
 namespace testing {
 
 using experimental::CertificateInfo;
 using experimental::Crl;
+using experimental::CrlImpl;
 using experimental::CrlProvider;
 
 TEST(CrlProviderTest, CanParseCrl) {
   std::string crl_string = GetFileContents(CRL_PATH);
   absl::StatusOr<std::shared_ptr<Crl>> result = Crl::Parse(crl_string);
   ASSERT_TRUE(result.ok());
   ASSERT_NE(*result, nullptr);
+  std::shared_ptr<CrlImpl> crl = std::static_pointer_cast<CrlImpl>(*result);
```

Comment:
We don't need to take another ref here.  Instead, we can just do this:

```
auto* crl = static_cast<CrlImpl*>(result->get());
```

Reply:
Changed